README.md

```diff
@@ -600,196 +390,98 @@ The PIN values MUST be within the UTF-8 character set. The PIN MUST be included
-    │                          │
-```
-
-### 3.5.1 Payload
+#### 5.3.1.2 Direct UCAN Challenge
 
-| Field  | Value                           | Description                                                    | Required |
-| ------ | ------------------------------- | -------------------------------------------------------------- | -------- |
-| `awv`  | `"0.1.0"`                       | AWAKE message version                                          | Yes      |
-| `type` | `"awake/msg"`                   | Generic AWAKE message type                                     | Yes      |
-| `mid`  | `sha256(resEcdhPk + reqEcdhPk)` | Message ID                                                     | Yes      |
-| `msg`  |                                 | Fulfilled challenge payload encrypted with Step 4 ECDH AES-key | Yes      |
+If UCAN auth is required by the Provider, the Requestor MUST provide a UCAN. This MUST be the same challenge strategy as the one set in the [Provider's payload][Provider challenge]: the UCAN MUST be encrypted with the session key and the IV from the enclosing payload, MUST be given in a raw format, and MUST be inline (without a JSON object wrapper or similar).
 
-#### 3.5.1.1 Encrypted Message
+The UCAN MUST be issued (`iss`) by the Requestor's DID (not the temporary DID), and its audience (`aud`) MUST be the Provider's DID. The `att` field MUST be set to an empty array (i.e. it MUST NOT delegate any capabilities). The `prf` array MUST fulfill the capabilities set by the Provider.
 
-The encrypted message payload MUST include an `awake/ack` field, with a value of the requestor's long-term DID. This payload MAY contain additional fields. This is often useful if dovetailing the ACK with the first message of the session.
+This MAY be used to prove that the Requestor has the same capabilities that the Requestor required from the Provider to start the handshake, such as when enforcing a minimum security clearance or proving functional equivalence between a single user's trusted devices.
 
-``` javascript
-{
-  ...,
-  "awake/ack": reqActualDid
-}
+```mermaid
+flowchart LR
+    subgraph Envelope [XChaCha-Poly1305 Envelope]
+        UCAN["UCAN
+            iss: RequestorActualDid,
+            aud: ProviderActualDid,&nbsp;
+            att: [],&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+            prf: ...&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+        "]
+    end
 ```
 
-## 4 Secure Session
+# 6. MLS Session
 
-```
-Requestor                  Responder
-    ⋮                          ⋮
-    │                          │
-    │         Messages         │ (6)
-    │◄────────────────────────►│
-    │                          │
-    ⋮                          ⋮
-```
+Proceed with MLS as normal.
```

Comment:
Should we add a few words on what steps MLS will undertake? I assume now MLS will exchange key material that is encrypted with our "actual" identifiers (ie. keys making the "actual" DIDs you've talked about in the spec here)?

In section 3, showing the overall sequence, you show the 'MLS Key Package' step, might be good to at least explain that?

Reply:
I can, but it's just repeating what's in the MLS spec. In theory, OpenMLS will do all of this for you.

Reply:
Yeah, a link to MLS spec section would be great. I assume you can't summarise what is roughly about to happen in the following steps in a few sentences?

Reply:
Well, I mean, the MLS protocol spec is over 6500 lines long 😅 

I can link to it in that section though 👍

Reply:
Done 👍